include/xen/be/XenGnttab.hpp

```diff
@@ -92,29 +92,33 @@ class XenGnttabBuffer
 public:
 
 	/**
-	 * @param[in] domId domain id
-	 * @param[in] ref   grant reference id
-	 * @param[in] prot  same flag as in mmap()
+	 * @param[in] domId  domain id
+	 * @param[in] ref    grant reference id
+	 * @param[in] prot   same flag as in mmap()
+	 * @param[in] offset offset to the data inside the buffer
 	 */
 	XenGnttabBuffer(domid_t domId, grant_ref_t ref,
-					int prot = PROT_READ | PROT_WRITE);
+					int prot = PROT_READ | PROT_WRITE,
+					int offset = 0);
 
 	/**
-	 * @param[in] domId domain id
-	 * @param[in] refs  array of grant reference ids
-	 * @param[in] count number of grant refgerence ids
-	 * @param[in] prot  same flag as in mmap()
+	 * @param[in] domId  domain id
+	 * @param[in] refs   array of grant reference ids
+	 * @param[in] count  number of grant refgerence ids
+	 * @param[in] prot   same flag as in mmap()
+	 * @param[in] offset offset to the data inside the buffer
 	 */
 	XenGnttabBuffer(domid_t domId, const grant_ref_t* refs, size_t count,
-					int prot = PROT_READ | PROT_WRITE);
+					int prot = PROT_READ | PROT_WRITE,
+					int offset = 0);
```

Comment:
Usage of 'int' for 'offset' will make possible to have negative offset. And this seems like possible error.
May be better to use unsigned type.

Reply:
+  Why don't just use size_t for offset?

Reply:
size_t sounds good